### ISSUE TITLE
bytes reversed in output

### ISSUE THREAD

**Issue by nkovacs** [Naguissa/uCRC16Lib#1]
According to a couple other implementations (https://crccalc.com/, http://www.sunshine2k.de/coding/javascript/crc/crc_js.html, github.com/howeyc/crc16), the crc value of "T" is supposed to be `0xE4D9` using the X-25 algorithm, which is apparently what this library implements. Yet your library returns `0xD9E4`. Is this a bug, or am I misunderstanding something?

**Comment by Naguissa** [comment]
It can be a bug due to byte ordering, specially on 8-bit mcus.

What mcu are you using?

**Comment by nkovacs** [comment]
Arduino Uno. I tried https://github.com/FrankBoesing/FastCRC too, it returns the correct result.

**Comment by Naguissa** [comment]
I'm suspecting about bit ordering on 8-bit mcu. Could you copy offendig code here?  Thanks.

**Comment by nkovacs** [comment]
```C
// FastCRC - Version: Latest 
#include <FastCRC.h>
#include <FastCRC_cpu.h>
#include <FastCRC_tables.h>

// uCRC16Lib - Version: Latest 
#include <uCRC16Lib.h>

FastCRC16 CRC16;

unsigned long currentMillis = 0;
unsigned long prevMillis = 0;
const unsigned long updateInterval = 5000;
const uint8_t hello[] = "Hello World";

void setup() {
    currentMillis = millis();
    prevMillis = currentMillis;
}

void send(const uint8_t* buffer, size_t size) {
    uint16_t crc = uCRC16Lib::calculate(buffer, size);
    uint16_t crc2 = CRC16.x25(buffer, size);
    Serial.println(crc);
    Serial.println(crc2);
}

void loop() {
    currentMillis = millis();
    if (currentMillis - prevMillis > updateInterval) {
        prevMillis = currentMillis;
        send(hello, sizeof(hello) / sizeof(uint8_t));
    }
}
```

**Comment by nkovacs** [comment]
Also, [your own example](https://github.com/Naguissa/uCRC16Lib/blob/master/examples/uCRC16Lib_example/uCRC16Lib_example.ino#L28) says `0xD9E4` is the correct crc for "T", while everything else says `0xE4D9`.

**Comment by Naguissa** [comment]
Hello,

I've found the problem:

For line transmission bytes are reversed. But not for standard operation.

So, I've changed it to do not perform that byte swap.


I'll update this repository, the "byte per byte" one and do corresponding releases in less than 2h; then in one hour more aprox. it will be available on Arduino IDE.


Thanks!

**Comment by Naguissa** [comment]
Done! It should be fixed in Arduino IDE in short time (will be 2.0.0 version)
